models/tasks.js

```diff
@@ -193,29 +194,45 @@ const fetchUserTasks = async (username, statuses = [], field, order) => {
   }
 };
 
-const getNewTask = async (skill, level) => {
-  const task = await tasksModel
-    .where("category", "==", skill)
-    .where("level", ">=", level)
-    .where("level", "<=", level + 2)
-    .where("status", "==", TASK_STATUS.AVAILABLE)
-    .limit(1)
-    .get();
+const getNewTask = async (skill = undefined, level = undefined) => {
+  const availableTasks = await tasksModel.where("status", "==", TASK_STATUS.AVAILABLE).get();
+  const idArray = [];
 
-  let taskData, id;
-  if (!task.empty) {
-    task.forEach((doc) => {
-      id = doc.id;
-      taskData = doc.data();
-    });
-    return {
-      task: {
-        id,
-        ...taskData,
-      },
-    };
+  let task;
+
+  if (!availableTasks.empty) {
+    availableTasks.forEach((item) => idArray.push(item.id));
+
+    if (!skill) {
+      task = await ItemModel.where("itemType", "==", "TASK").where("levelNumber", "<=", 2).get();
+    } else {
+      task = await ItemModel.where("tagName", "==", skill)
+        .where("itemType", "==", "TASK")
+        .where("levelNumber", ">=", level)
+        .where("levelNumber", "<=", level + 2)
+        .get();
+    }
   }
 
+  if (!task.empty) {
+    let taskData, id;
+    for (const doc of task.docs) {
+      if (idArray.includes(doc.data().itemId)) {
+        id = doc.id;
+        taskData = doc.data();
+        break;
+      }
+    }
+    if (taskData) {
+      return {
+        task: {
+          id,
+          ...taskData,
+        },
+      };
+    }
+    return { taskNotFound: true };
+  }
   return { taskNotFound: true };
```

Comment:
Task not found two time?

Reply:
yes if there is no available task then also we want to return that, and if there was available but we couldn't find anything we will return taskNotFound